src/main/kotlin/io/handicrafty/sixteen/notification/model/Notification.kt

```diff
@@ -0,0 +1,57 @@
+package io.handicrafty.sixteen.notification.model
+
+import io.handicrafty.sixteen.common.jpa.enumeration.NotificationType
+import io.handicrafty.sixteen.common.jpa.model.CreateTimeAuditEntity
+import io.handicrafty.sixteen.user.model.User
+import jakarta.persistence.*
+import jakarta.validation.constraints.NotBlank
+import org.hibernate.Hibernate
+import org.hibernate.annotations.JdbcTypeCode
+import org.hibernate.type.SqlTypes
+
+@Entity
+@Table(name = "notification")
+class Notification(
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", nullable = false)
+    var id: Long? = null,
+
+    @ManyToOne(optional = false, fetch = FetchType.LAZY)
+    @JoinColumn(name = "user_id", nullable = false)
+    var user: User,
+
+    @field:NotBlank(message = "제목은 비어있을 수 없습니다.")
+    @Column(name = "title", nullable = false, length = 255)
+    var title: String,
+
+    @field:NotBlank(message = "내용은 비어있을 수 없습니다.")
+    @Column(name = "body", nullable = false, length = 5000)
+    var body: String,
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "notification_type", nullable = false, length = 50)
+    var notificationType: NotificationType,
+
+    @JdbcTypeCode(SqlTypes.JSON)
+    @Column(name = "metadata", nullable = false)
+    var metadata: Map<String, Any> = emptyMap(),
+) : CreateTimeAuditEntity() {
+    @Column(name = "user_id", nullable = false, insertable = false, updatable = false)
+    private val userId: Long? = null
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
+        other as Notification
+
+        return id != null && id == other.id
+    }
+
+    override fun hashCode(): Int = javaClass.hashCode()
+
+    @Override
+    override fun toString(): String {
```

Comment:
nit: 재정의 할 수 없도록 이런 형태를 제안드려봅니다!
```suggestion
    final override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
        other as Notification

        return id != null && id == other.id
    }

    final override fun hashCode(): Int = javaClass.hashCode()

    final override fun toString(): String {
```

Reply:
수정했습니다. https://github.com/handicrafty/sixteen-backend/pull/14/commits/454419f242c301c1c11e55e7dd92648523f57547